### PR TITLE
feature/swap

### PR DESCRIPTION
**Pull request by SteveMieskoski**
### Feature
- [x] Updated CHANGELOG.md  
- [ ] Add PR label

### CODE REVIEW

File: src/partners/partnersConfig/OtherChains.js [hunk 1]
```diff
@@ -0,0 +1,411 @@
+export default {
```
Comment (gamalielhere) [inline src/partners/partnersConfig/OtherChains.js]: Maybe move this to nodeTypes and add a key `supportsSwap`.

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 2]
```diff
@@ -115,54 +162,482 @@
...
+  bestRateForQuantity,
+  isValidEntry,
+  providerNames,
+  BASE_CURRENCY,
+  MIN_SWAP_AMOUNT
+} from '@/partners';
+
+const errorLogger = debug('v5:swapContainer');
+
+BigNumber.config({ DECIMAL_PLACES: 6 });
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: removed, and changed use cases to use .toFixed()

File: src/partners/kyber/kyber.js [hunk 3]
```diff
@@ -285,87 +184,168 @@ export default class Kyber {
...
-    );
-    return await contract.methods
-      .allowance(userAddress, this.getKyberNetworkAddress())
-      .call();
+  approveKyber(fromToken, fromValueWei) {
+    try {
+      return {
+        to: this.getTokenAddress(fromToken),
+        value: 0,
+        data: new this.web3.eth.Contract(
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/partners/kyber/kyber.js]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/components/SwapConfirmationModal/SwapConfirmationModal.vue [hunk 4]
```diff
@@ -89,206 +95,136 @@ export default {
...
+          swapDetails.fromCurrency !== BASE_CURRENCY
+        ) {
+          const tokenInfo = EthereumTokens[swapDetails.fromCurrency];
+          if (!tokenInfo) throw Error('Selected Token not known to MEW Swap');
+
+          this.preparedSwap = {
+            from: this.$store.state.wallet.getChecksumAddressString(),
+            to: tokenInfo.contractAddress,
+            value: 0,
+            data: new this.web3.eth.Contract(
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/components/SwapConfirmationModal/SwapConfirmationModal.vue]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/components/SwapConfirmationModal/SwapConfirmationModal.vue [hunk 5]
```diff
@@ -89,206 +95,136 @@ export default {
...
+      ) {
+        if (
+          swapDetails.maybeToken &&
+          swapDetails.fromCurrency !== BASE_CURRENCY
+        ) {
+          const tokenInfo = EthereumTokens[swapDetails.fromCurrency];
+          if (!tokenInfo) throw Error('Selected Token not known to MEW Swap');
+
+          this.preparedSwap = {
+            from: this.$store.state.wallet.getChecksumAddressString(),
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/components/SwapConfirmationModal/SwapConfirmationModal.vue]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 6]
```diff
@@ -115,54 +162,482 @@
...
+        (isValidEntry(this.fromValue) && to === 'from') ||
+        (isValidEntry(this.toValue) && to === 'to')
+      ) {
+        if (
+          this.swap.getProvider(this.providerNames.simplex).currencies.fiat[
+            this.fromCurrency
+          ]
+        ) {
+          this.web3.utils._.debounce(
+            this.updateEstimate(this.providerNames.simplex + to),
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: I changed the variable name to be more descriptive.

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 7]
```diff
@@ -55,57 +92,67 @@
...
-          </div>
-          <div class="small-circle-button-green-border active">
-            {{ $t('common.regular') }}
-          </div>
-          <div class="small-circle-button-green-border">
-            {{ $t('common.fast') }}
-          </div>
+        <div class="title title-and-copy">
+          <h4>
+            {{ fromCurrency }} address for refund (if needed) from Changelly
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 8]
```diff
@@ -1,11 +1,27 @@
...
+      :current-address="currentAddress"
+      @swapStarted="resetSwapState"
+    />
 
     <div class="title-block">
       <interface-container-title :title="$t('common.swap')" />
       <div class="buy-eth">
-        <span>Buy ETH with</span> <img :src="images.visaMaster" />
+        <a href="https://ccswap.myetherwallet.com" target="_blank">
+          <span>Buy ETH with</span> <img :src="images.visaMaster" />
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 9]
```diff
@@ -115,54 +162,482 @@
...
+          }
+          return bestRateForQuantity([...this.providerList], this.fromValue);
+        }
+      } catch (e) {
+        errorLogger(e);
+      }
+    },
+    fromBelowMinAllowed() {
+      if (MIN_SWAP_AMOUNT > +this.fromValue)
+        return `Below minimun swap value of ${MIN_SWAP_AMOUNT}`;
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 10]
```diff
@@ -115,54 +162,482 @@
...
+        );
+      } else if (this.fromCurrency === this.baseCurrency) {
+        return +this.fromValue >= this.account.balance;
+      }
+      return false;
+    }
+  },
+  watch: {
+    ['this.$store.state.network.type.name']() {
+      this.swap.updateNetwork(this.$store.state.network.type.name);
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 11]
```diff
@@ -115,54 +162,482 @@
...
+    },
+    async getBalance(currency) {
+      if (this.swap.isToken(currency) && currency !== this.baseCurrency) {
+        const balance = await new this.web3.eth.Contract(
+          [
+            {
+              constant: true,
+              inputs: [
+                {
+                  name: '_owner',
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: fixed

File: src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue [hunk 12]
```diff
@@ -115,54 +162,482 @@
...
+          )
+        );
+      } else if (this.fromCurrency === this.baseCurrency) {
+        return +this.fromValue >= this.account.balance;
+      }
+      return false;
+    }
+  },
+  watch: {
+    ['this.$store.state.network.type.name']() {
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/layouts/InterfaceLayout/containers/SwapContainer/SwapContainer.vue]: fixed

File: src/containers/ConfirmationContainer/components/ConfirmCollectionModal/ConfirmCollectionModal.vue [hunk 13]
```diff
@@ -189,6 +201,11 @@ export default {
       return 0;
     }
   }
+  // methods:{
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/containers/ConfirmationContainer/components/ConfirmCollectionModal/ConfirmCollectionModal.vue]: fixed

File: src/partners/simplex/simplex.js [hunk 14]
```diff
@@ -44,98 +58,170 @@ export default class Simplex {
     return !!this.currencies.fiat[currency];
   }
 
+  // ============================= pair and value selection and update methods  ====================================
```
Comment (kvhnuke) [inline src/partners/simplex/simplex.js]: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/

File: src/partners/simplex/simplex.js [hunk 15]
```diff
@@ -44,98 +58,170 @@ export default class Simplex {
...
+    this.currentOrder = result.result;
     return {
-      fromValue: rate.result.fiat_money.base_amount,
-      toValue: rate.result.digital_money.amount
+      fromValue: result.result.fiat_money.base_amount,
+      toValue: result.result.digital_money.amount
     };
   }
 
+  // ============================= Determine inclusion in currency options ====================================
```
Comment (kvhnuke) [inline src/partners/simplex/simplex.js]: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/

File: src/partners/simplex/simplex.js [hunk 16]
```diff
@@ -44,98 +58,170 @@ export default class Simplex {
...
+            collectMap.set(prop, {
+              symbol: prop,
+              name: this.currencies.digital[prop].name
+            });
+        }
+      }
+    }
+  }
+
+  // ============================= Finalize swap details ====================================
```
Comment (kvhnuke) [inline src/partners/simplex/simplex.js]: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/

File: src/partners/kyber/config/config.js [hunk 17]
```diff
@@ -1,8 +1,3 @@
-const kyberAddressFallback = {
-  ETH: '0x818e6fecd516ecc3849daf6845e3ec868087b755',
-  ROP: '0x91a502C678605fbCe581eae053319747482276b9'
-};
-
 // TODO: look into only using kyberTokenInfoList for ETH and parsing results differently if ROP?
```
[the remark this answers is not in the document]
Comment (SteveMieskoski) [reply src/partners/kyber/config/config.js]: I agree,  for the moment the ropsten version is useful for development.  We could introduce a check for production so is doesn't show up, but remains for development?

**Pull request merged**